File: src/main/scala/scalainterpreter/DisplayWatchedVars.scala


package  scalainterpreter

object DisplayWatchedVars {

  def watchVar(varName: String) = {
    var valOfVar = scalaExec.Interpreter.GlobalValues.globalInterpreter.valueOfTerm(varName)
    var typeOfVar = scalaExec.Interpreter.GlobalValues.globalInterpreter.typeOfTerm(varName)
    
    //println("var = "+varName + " has value = "+valOfVar)
  }
}

File: src/main/java/scalainterpreter/displayWatchedVars.java
package scalainterpreter;

import scala.Option;
import scalaExec.Interpreter.GlobalValues;

public class displayWatchedVars {

    public static void watchVar(String varName)  {
        Option<Object>  valOfVar = GlobalValues.globalInterpreter.valueOfTerm(varName);
        System.out.println("watching var = "+varName);
        javax.swing.JOptionPane.showMessageDialog(null, valOfVar);
    }
}
